src/main/java/com/couchbase/lite/storage/JavaSQLiteStorageEngine.java

```diff
@@ -1,514 +1,289 @@
-            // some more works after interrupting the thread (See ChangeTracker.stop()).
-            Thread.interrupted();
-
-            sem.acquire();
-
-            if (primaryConnection == null)
-                sem.release();
-
-            return primaryConnection;
-        }
-
-        private void release(Connection conn) {
-            sem.release();
-        }
-
-        protected Connection getConnection(boolean markUsage) {
-            if (!isStart.get()) {
-                throw new IllegalStateException("Cannot use the pool without starting the pool.");
-            }
-
-            Connection conn = threadLocal.get();
-            if (conn == null) {
-                try {
-                    conn = acquire();
-                } catch (Exception e) {
-                    throw new RuntimeException(e);
-                }
-
-                threadLocal.set(conn);
-            }
-
-            if (markUsage)
-                conn.usageCount++;
-
-            return conn;
-        }
-
-        public Connection getConnection() {
-            return getConnection(true);
-        }
+    private static final boolean SUPPORT_ENCRYPTION = true;
 
-        public void releaseConnection(Connection conn) {
-            if (!isStart.get()) {
-                throw new IllegalStateException("Cannot use the pool without starting the pool.");
-            }
-
-            if (conn == null) return;
-
-            conn.usageCount--;
-
-            if (conn.usageCount == 0) {
-                threadLocal.set(null);
-                release(conn);
-            }
-        }
-    }
-
-    private String path;
+    private SQLiteDatabase database;
 
-    static {
-        NativeUtils.loadLibrary("CouchbaseLiteJavaNative");
+    public JavaSQLiteStorageEngine() {
+        // Do Nothing
     }
 
-    // Currently this is used by the native library to throw a SQLException back to Java
-    private static void throwSQLException(String msg) throws SQLException {
-        throw new SQLException(msg);
-    }
-
-    private native long _open(String path);
-
     @Override
-    public synchronized boolean open(String path, String encryptionKey) {
-        this.path = path;
-        connectionPool.start();
-        return true;
+    public boolean open(String path) throws SQLException {
+        if(database != null && database.isOpen())
+            return true;
+        try {
+            database = SQLiteDatabase.openDatabase(path, null, SQLiteDatabase.CREATE_IF_NECESSARY);
+            Log.v(Log.TAG_DATABASE, "%s: Opened Android sqlite db", this);
+        } catch(Throwable e) {
+            Log.e(Log.TAG_DATABASE, "Unable to open the SQLite database", e);
+            if (database != null)
+                database.close();
+            throw new SQLException(e);
+        }
+        return database.isOpen();
     }
 
-    private native int _getVersion(long handle);
-
     @Override
     public int getVersion() {
```

Comment:
This is general comments for all methods that are declared in this class.
- As all method could throw SQLException, but method signature is not defined with throws SQLException. Is it correct?